engine/src/TimeSystem.cpp

```diff
@@ -0,0 +1,67 @@
+#include "Engine/Core/TimeSystem/TimeSystem.hpp"
+#include "Engine/Core/Debug/Log.hpp"
+#include <string>
+#include <thread> // std::this_thread::sleep_for
+
+using namespace Engine::Core;
+using namespace Engine::Core::Debug;
+
+void TimeSystem::update(std::function<void(double fixedUnscaledDetlaTime, double fixedDetlaTim)> fixedUpdateFunction,
+                        std::function<void(double unscaledDeltaTime, double deltaTime)> updateFunction,
+                        std::function<void()> renderFunction) noexcept
+{
+    /*First render the current frame*/
+    renderFunction();
+
+    /*Second, prepar the next frame*/
+    m_tempTime = std::chrono::system_clock::now();
+    m_unscaledDeltaTime = std::chrono::duration<double>(m_tempTime - m_time).count();
+    m_time = m_tempTime;
+
+    /*Apply time scale*/
+    m_deltaTime = m_unscaledDeltaTime * m_timeScale;
+
+    /*Add accumulator*/
+    m_scaledTimeAcc += m_deltaTime;
+    m_unscaledTimeAcc += m_unscaledDeltaTime;
+
+    m_scaledTimeAcc *= !isinf(m_scaledTimeAcc);     // reset if isInf (avoid conditionnal jump)
+    m_unscaledTimeAcc *= !isinf(m_unscaledTimeAcc); // reset if isInf (avoid conditionnal jump)
+
+    /*Fixed update*/
+    m_fixedTimeAcc += m_unscaledDeltaTime;
+
+    while (m_fixedTimeAcc >= m_fixedUnscaledDetlaTime)
+    {
+        fixedUpdateFunction(m_fixedUnscaledDetlaTime, m_fixedDetlaTime);
+        m_fixedTimeAcc -= m_fixedUnscaledDetlaTime;
+    }
```

Comment:
I'm not sure I fully understand what this is doing, I'm sorry if I get this wrong: `m_fixedTimeAcc` is incremented by `m_unscaledDeltaTime` every time `TimeSystem::update()` is called, but it is then compared to `m_fixedUnscaledDetlaTime`. Shouldn't it be compared to `m_fixedDeltaTime` instead?